Assets/Scripts/Player/PlayerState.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+using Random = System.Random;
+
+namespace Player
+{
+    
+    [System.Serializable] public class UnityEventInt:UnityEvent<int> {}
+    public class PlayerState : MonoBehaviour
+    {
+        private readonly Random _random = new Random();
+        
+        //Player State values
+        [Tooltip("100 - full health, 0 - dead")] [SerializeField] [Range(0, 100)] 
+        private int health;
+
+        [Tooltip("100 means no hunger, 0 means the player is in desperate need for food")] [SerializeField] [Range(0, 100)] 
+        private int hunger;
+
+        [Tooltip("100: not thirsty, 0: gazing for a sip of water")] [SerializeField] [Range(0, 100)] 
+        private int thirst;
+        
+        [Header("Event that will throw when the player's health changes")]
+        public UnityEventInt playerHeathUpdated;
+
+        [Header("Event that will throw when the player's hunger changes")]
+        public UnityEventInt playerHungerUpdated;
+        
+        [Header("Event that will throw when the player's thirst changes")]
+        public UnityEventInt playerThirstUpdated;
+        
+        //Interface
+        public void ChangePlayerHealth(int changeBy)
+        {
+            int updatedValue = health + changeBy;
+            if (updatedValue > 100) updatedValue = 100;
+            else if (updatedValue < 0)
+            {
+                updatedValue = 0;
+                //TODO: invoke death
+            }
+
+            health = updatedValue;
+            
+            //throws an event with the new health value as a parameter
+            playerHeathUpdated.Invoke(updatedValue);
+        }
+
+        public void ChangePlayerHunger(int changeBy)
+        {
+            int updatedValue = hunger + changeBy;
+            if (updatedValue > 100) updatedValue = 100;
+            else if (updatedValue < 0)
+            {
+                updatedValue = 0;
+                ChangePlayerHealth(-1);
+            }
+
+            hunger = updatedValue;
+            playerHungerUpdated.Invoke(updatedValue);
+        }
+
+        public void ChangePlayerThrist(int changeBy)
+        {
+            int updatedValue = thirst + changeBy;
+            if (updatedValue > 100) updatedValue = 100;
+            else if (updatedValue < 0)
+            {
+                updatedValue = 0;
+                ChangePlayerHealth(-1);
+            }
+
+            thirst = updatedValue;
+            playerThirstUpdated.Invoke(updatedValue);
+        }
+    }
```

Comment:
I think a positive value for heal and a negative value for damage is clear enough. The methods are small enough and do not need to be broken up for maintainability reasons.

Reply:
no changes for now

Reply:
I agree with Jan. I think an interface IDamageable would be handy. If we make PlayerState implement IDamageable, we should add another method like Hit(int damage) which itself calls ChangePlayerHealth with (-damage). Also you could give the interface a method Heal(int amount) which does the same just with healing, because damageables should also be able to be healed, so this fits in here.

Also see implementation of attack system.

Reply:
+ ChangePlayerHealth = private ;)